ci_framework/roles/operator_build/tasks/build.yml

```diff
@@ -14,17 +14,29 @@
 # License for the specific language governing permissions and limitations
 # under the License.
 
-- name: "{{ operator.name }} - Set default api path"
-  ansible.builtin.set_fact:
-    operator_api_path: "github.com/{{ cifmw_operator_build_org }}/{{ operator.name }}/api"
+- name: "{{ operator.name }} - Get base module name and set the api path"
+  when:
+    - cifmw_operator_build_meta_build
+    - operator.name != cifmw_operator_build_meta_name
+  block:
+    - name: "{{ operator.name }} - Get base module name from go.mod"
+      ansible.builtin.shell: |
+        grep {{ cifmw_operator_build_org }}/{{ operator.name }}/api go.mod | head -n 1 | xargs basename
```

Comment:
I agree with @marios comment on the review meeting, why not use Ansible? This can be perfectly done with lineinfile module and a regex such as this one (example, should be parameterized obviously) `github\.com\/openstack-k8s-operators\/infra-operator\/(\w*)\s`

Reply:
ack - but can be done in a follow up if this is blocking

Reply:
I don't have a preference here, I think that the shell was a quick solution to have a fix landing asap.
If it is something that must merge asap, we can think on a follow up?

Reply:
Sure, please, create a subtask here https://issues.redhat.com/browse/OSP-25624

Reply:
I will fix it here instead of adding another card to the backlog. 

Just for my understanding, @marios @pablintino could you please elaborate your reasoning why do you think ansible way is better? So that next time I will prefer one over the other.

Reply:
Is just best practices, here you have and intro: https://www.ansible.com/blog/ansible-best-practices-essentials (point four). There they talk about error management, but we can talk about idempotence too as most modules properly compute the `changed` property, about consistent run in multiple environments, here the use case is very basic, but think about commands that behaves differently between versions, the exact same role will behave differently, but, with native modules, as all is done in the ansible implementation of the module and versions are fixed, the results of the module run will be equal. We have tried the most to use native modules because of those reasons, so, if that reasons are not enough, we can talk about consistency too, as we are trying to use them whenever possible, why not try to apply that rule everywhere when possible?

Reply:
I discussed this with Pablo, I read more about lineinfile but all the example I found on internet are either adding/deletion/replacing things and not just search which I wanted, looks like the `lineinfile` module is primarily used for modifying lines in a file rather than searching.

https://www.middlewareinventory.com/blog/ansible-lineinfile-examples/

we have created https://issues.redhat.com/browse/OSP-26014 to improve this in a followup. lets merge this as it is and I will improve this in a followup.

Reply:
Thanks Pablo for your earlier explanation and link why using ansible module is better and for suggesting the alternative module.

I have updated the code.